### PR TITLE
Move all files from "pc" to "hosted" and rename "cl_utils.* to "cli.*.

### PR DESCRIPTION
**Pull request by sidprice**
Built and tested on Windows 11, not built for Linux.

### CODE REVIEW

**Review by dragonmux — CHANGES_REQUESTED** [review]
There's only one thing we spotted - this otherwise look great. This will be ready for merge once that one item is addressed.

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,5 @@ blackmagic_upgrade
 .vscode
 .gdb_history
 src/artifacts/
+
+refs/
```
[the remark this answers is not in the document]
Comment (sidprice) [reply .gitignore:24]: The refs folder contains the text-based help files I have been preparing for the update of BMDA "-h" option.

[the remark this answers is not in the document]
Comment (dragonmux) [reply .gitignore:24]: ah, ok.. given that's something local to you and will never see the repo, please remove this ignore from the patchset (you can have it locally, that's understandable and fine, just.. not in repo)

**Comment by sidprice** [comment]
@dragonmux Sure, however, I don't know how to do that, i.e. maintain a local .gitignore?

**Comment by dragonmux** [comment]
To accomplish this, what you'll need to do is first remove the change to the ignore, `git add .gitignore`, then commit it as a fixup to the commit it came in on (`git commit --fixup 81b804a`) - once you've done that you can edit the change back in but leave it not-added (`git add -a` is frowned on anyway so as long as you don't use that to add changes, it won't be a problem).

Once all that's done, `git rebase -i master --autosquash` to apply the fixup and then `git push -f` to update the PR.

**Comment by sidprice** [comment]
Sounds awfully complicated ... why don't I just fix .gitignore and move my refs folder outside the repo?

**Comment by dragonmux** [comment]
That'd be fine, though the fixup process is similar (the commit and rebase probably seem scary, but they're really not that bad and Git'll guide you through it)

**Comment by dragonmux** [comment]
Please rebase the gitignore fix as a fixup commit to 81b804a as this introduces noise which can be easily avoided. `git rebase -i master` and move efc5b64 just under 81b804a in the text editor, changing the 'pick' prefix to 'f'. A force-push will be required after.

**Comment by sidprice** [comment]
I am way out of my depth with git here, quite lost.

**Comment by dragonmux** [comment]
That problem is why we suggested using a fixup commit as then the tools automate things - `git reset HEAD~` to go back one commit, `git add .gitignore` to stage the ignore fix, `git commit --fixup 81b804a` to commit it as a fixup for 81b804a that can be automatically applied, then `git rebase master --autosquash` to automatically apply it. `git push -f` to force push on completion - please trust us that this is correct and will work. Exact command lines to run given

**Comment by sidprice** [comment]
OK:

PS D:\DataRoot\Projects\blackmagic> git reset HEAD                           
PS D:\DataRoot\Projects\blackmagic> git add .gitignore
PS D:\DataRoot\Projects\blackmagic> git commit --fixup 81b804a
On branch BMDA_CLI_help
Your branch is up to date with 'origin/BMDA_CLI_help'.

nothing to commit, working tree clean
PS D:\DataRoot\Projects\blackmagic>

**Comment by dragonmux** [comment]
you missed the ~ off HEAD.

**Comment by sidprice** [comment]
OK, seems better :)

**Review by dragonmux — DISMISSED** [review]
LGTM. We saw the CI build passed so we'll merge this.

**Comment by dragonmux** [comment]
@sidprice went to go merge this and spotted that the `git rebase -i master --autosquash` step hadn't been done, so the fixup commit is still present.

Could we get you to run that step and do a `git push -f` please? Once we've seen that happen, we'll merge this.

**Review by dragonmux — APPROVED** [review]
LGTM, thank you sid! Merging.

**Pull request merged**